fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -894,12 +894,17 @@ public void executeByLegacy(TUniqueId queryId) throws Exception {
                     && !(parsedStmt instanceof TransactionStmt)) {
                 throw new TException("This is in a transaction, only insert, commit, rollback is acceptable.");
             }
+            SessionVariable sessionVariable = context.getSessionVariable();
+            if (!(parsedStmt instanceof SetStmt) && !(parsedStmt instanceof UnsetVariableStmt)) {
```

Comment:
why need except set and unset?

Reply:
the unset stmt will be reverted because `disableNereidsPlannerOnce`